cuda_core/cuda/core/experimental/_device.pyx

```diff
@@ -1095,7 +1107,10 @@ class Device:
         # Use 256 characters to be consistent with CUDA Runtime
         cdef int LENGTH = 256
         cdef bytes name = bytes(LENGTH)
-        HANDLE_RETURN(cydriver.cuDeviceGetName(<char*>name, LENGTH, self._id))
+        cdef char* name_ptr = name
+        cdef cydriver.CUdevice this_dev = self._id
+        with nogil:
```

Comment:
Nit: I'd expect that nogil annotations would be used at the parent scope (to the cython function) b/c we wanted to batch more functions into the nogil scope.  Instead of wrapping a single function with nogil. Is there a feature in Cython where you can turn off the gil in the wrapping function so you don't need to push the annotation into the parent scope?